src/components/App.js

```diff
@@ -59,6 +58,7 @@ class App extends Component {
         console.log("hey, I'm the infoWindow");
         this.monitorCurrentLocation();
         monitorInterval = setInterval(this.monitorCurrentLocation, 1000 * 10);
```

Comment:
One thing to note is before we assign a value to `monitorInterval`, we should ensure the interval referenced by `monitorInterval` has been stopped. 

```
if (monitorInterval !== undefined) {
  clearInterval(monitorInterval);
  monitorInterval = undefined;
}
monitorInterval = setInterval(this.monitorCurrentLocation, 1000 * 10);	
```

Reply:
Thank you this really helps and I believe this makes sense! Next time we meet I'd like to go over it just to make sure I get all the nuances of why this is need in this way.